WooCommerce/Classes/Extensions/UITableViewDiffableDataSource+Helpers.swift

```diff
@@ -0,0 +1,34 @@
+import UIKit
+
+/// Additional utilities for `UITableViewDiffableDataSource`.
+///
+@available(iOS 13.0, *)
+extension UITableViewDiffableDataSource {
+    /// Returns the index of the item pointed to by `indexPath` relative to the number of
+    /// items in `snapshot()`.
+    func indexOfItem(for indexPath: IndexPath) -> Int? {
+        guard let identifier = itemIdentifier(for: indexPath) else {
+            return nil
+        }
+
+        return snapshot().indexOfItem(identifier)
```

Comment:
[Q] Would this be expensive to access?

Reply:
I was concerned about this too. I tested it with an order list that has more than 1K orders. I used an iPhone 8, the slowest device that I have. It doesn't seem to cause any issues in scrolling. 

The [`snapshot()`](https://developer.apple.com/documentation/uikit/uitableviewdiffabledatasource/3255156-snapshot) returns an `NSDiffableDataSourceSnapshot` which hints that it is a `struct`. But then there is also an [Objective-C equivalent of `NSDiffableDataSourceSnapshot`](https://developer.apple.com/documentation/uikit/nsdiffabledatasourcesnapshot?language=objc) which is a `class`. I wonder if the `struct` is just bridged to the `class` and we're simply using a reference here. 🤔 That could mean that this is fast. 

In any case, if this becomes a problem, an option for us is to preserve a reference to the snapshot during `apply`:

https://github.com/woocommerce/woocommerce-ios/blob/5631e4e1f86a10ead37430d6d8c60f9885aea1be/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L201-L204

We'd have to do wrap that with something to reduce duplication.